index.js

```diff
@@ -12,7 +12,63 @@ function setMultipart (req, done) {
   done()
 }
 
+function attachToBody (options, req, reply, next) {
+  req[kMultipart] = true
+
+  const body = { }
+  const mp = req.multipart((field, file, filename, encoding, mimetype) => {
+    body[field] = {
+      // stream: file,
+      data: [],
+      filename,
+      encoding,
+      mimetype,
+      limit: false
+    }
+
+    const fileData = []
+    const isCustomDataConsumer = typeof options.onData === 'function'
+    const onData = isCustomDataConsumer
+      ? options.onData
+      : (fieldName, data) => { fileData.push(data) }
+
+    file.on('data', (data) => { onData(field, data) })
```

Comment:
I’m not convinced we should do this. In this way we do not support backpressure at all: it is going to cause issues. I think we should just remove the onData option.

Reply:
Then I don't understand how we could consume the stream: we put the chunk in an array as was the default? Or are you thinking another way?
Maybe the user could pass as argument a Stream Writer Factory where we will pipe the input stream?

Reply:
`onData` seems to cater for the advanced use case (otherwise I would keep the default), but **`onData` does not support backpressure**:  an advanced user will not be able to use it. If you want to `attachToBody`, we can assume you want to take the whole file in memory.

One way to handle this use case is to have an `onFile` callback, where a stream is passed in. An implementer would have to guarantee that the stream is fully consumed.

> Then I don't understand how we could consume the stream: we put the chunk in an array as was the default? 

Yes essentially.